library/cartridge_check_instance_started.py

```diff
@@ -4,32 +4,93 @@
 from ansible.module_utils.helpers import ModuleRes, CartridgeException
 from ansible.module_utils.helpers import get_control_console
 
-
 argument_spec = {
     'control_sock': {'required': True, 'type': 'str'},
     'stateboard': {'required': True, 'type': 'bool'},
+    'expected_state': {'required': False, 'type': 'list', 'default': ['Unconfigured', 'RolesConfigured']},
+    'check_buckets_discovery': {'required': False, 'type': 'bool', 'default': False},
 }
 
 
-def check_instance_started(params):
-    try:
-        control_console = get_control_console(params['control_sock'])
-        instance_is_alive = True
-        if not params['stateboard']:
-            instance_is_alive, _ = control_console.eval_res_err('''
-                return require('membership').myself().status == 'alive'
+def check_stateboard_started(control_console):
+    box_status, err = control_console.eval_res_err('''
+        if type(box.cfg) == 'function' then
+            return nil, "Box hasn't been configured!"
+        end
+        return box.info.status
+    ''')
+    if not box_status:
+        return ModuleRes(success=False, msg="Impossible to get stateboard status: %s!" % err)
+    if box_status != 'running':
+        return ModuleRes(success=False, msg="Stateboard is not running, it has '%s' status!" % box_status)
+
+    return ModuleRes(success=True)
+
+
+def check_instance_started(control_console, expected_state, check_buckets_discovery):
+    member_status, err = control_console.eval_res_err("return require('membership').myself().status")
+    if not member_status:
+        return ModuleRes(success=False, msg="Impossible to get membership status: %s!" % err)
+    if member_status != 'alive':
+        return ModuleRes(success=False, msg="Membership is not alive, it has '%s' status!" % member_status)
+
+    cartridge_state, err = control_console.eval_res_err("return require('membership').myself().payload.state")
+    if not cartridge_state:
+        return ModuleRes(success=False, msg="Impossible to get instance Cartridge state: %s!" % err)
+    if cartridge_state not in expected_state:
+        return ModuleRes(
+            success=False,
+            msg="Instance is not in one of states: %s, it's in '%s' state!" % (
```

Comment:
Why your errors are so emotional?) You actively use `!`.

Reply:
Because I think that with `.` are informational messages, and with `!` - error messages. But I removed the `!` from all messages so they don't look threatening